plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -264,8 +265,26 @@ func getHTTPRoundTripper(c *config.Configuration, logger *zap.Logger) (rt http.R
 		TLSHandshakeTimeout: 10 * time.Second,
 		TLSClientConfig:     ctlsConfig,
 	}
+
+	token := ""
+	if c.TokenFilePath != "" {
+		tokenFromFile, err := loadToken(c.TokenFilePath)
+		if err != nil {
+			return nil, err
```

Comment:
We'll need to add a couple of tests to cover the error cases here and on L287.

Reply:
@albertteoh To cover these i was thinking of adding back table test case [here](https://github.com/ChillOrb/jaeger/blob/2c56f164f8a40f4a9d8d901753131454bc5abe73/plugin/metrics/prometheus/metricsstore/reader_test.go#L361) and [here](https://github.com/ChillOrb/jaeger/blob/2c56f164f8a40f4a9d8d901753131454bc5abe73/plugin/metrics/prometheus/metricsstore/reader_test.go#L505) 

I was thinking of using `wantErr bool` field and to mock an error maybe i can mock the behavior where user provides incorrect file path ( path does not exist error). This would however add some clumsy `if else` blocks where i check `if wantErr{ require.Err()}else{require.NoErr()}`.

Does this sound ok?

Reply:
I think a better approach is to add a seperate simple test to cover the error case. This should cover both lines:
```
func TestGetRoundTripperTokenError(t *testing.T) {
	tokenFilePath := "this file does not exist"

	_, err := getHTTPRoundTripper(&config.Configuration{
		TokenFilePath: tokenFilePath,
	}, nil)

	require.Error(t, err)

	pathErr := &os.PathError{}
	assert.ErrorAs(t, err, &pathErr)
	assert.EqualError(t, pathErr, fmt.Sprintf("open %s: no such file or directory", tokenFilePath))
}
```

Reply:
@albertteoh thanks, i was overthinking this